cognite/client/_api/functions.py

```diff
@@ -931,41 +932,53 @@ def create(
+            data (optional, Dict): Data to be passed to the scheduled run.
 
         Returns:
             FunctionSchedule: Created function schedule.
 
+        Warning:
+            Do not pass secrets or other confidential information via the ``data`` argument. There is a dedicated
+            ``secrets`` argument in FunctionsAPI.create() for this purpose.
+
         Examples:
 
-            Create function schedule::
+            Create a function schedule that runs using specified client credentials (**recommended**)::
 
+                >>> import os
                 >>> from cognite.client import CogniteClient
                 >>> from cognite.client.data_classes import ClientCredentials
                 >>> c = CogniteClient()
                 >>> schedule = c.functions.schedules.create(
-                ...     name= "My schedule",
+                ...     name="My schedule",
                 ...     function_id=123,
                 ...     cron_expression="*/5 * * * *",
-                ...     client_credentials=ClientCredentials("my-client-id", "my-client-secret"),
-                ...     description="This schedule does magic stuff."
+                ...     client_credentials=ClientCredentials("my-client-id", os.environ["MY_CLIENT_SECRET"]),
+                ...     description="This schedule does magic stuff.",
+                ...     data={"magic": "stuff"},
                 ... )
 
-        """
-        _get_function_identifier(function_id, function_external_id)
+            You may also create a schedule that runs with your -current- credentials, i.e. the same credentials you used
+            to instantiate the ``CogniteClient`` (that you're using right now). **Note**: Unless you happen to already use
+            client credentials, *this is not a recommended way to create schedules*, as it will create an explicit dependency
+            on your user account, which it will run the function "on behalf of" (until the schedule is eventually removed)::
 
-        nonce = None
-        if client_credentials is not None:
-            if function_id is None:
-                raise ValueError("When passing 'client_credentials', 'function_id' must be set")
-            nonce = _create_session_and_return_nonce(self._cognite_client, client_credentials)
+                >>> schedule = c.functions.schedules.create(
+                ...     name="My schedule",
+                ...     function_id=456,
+                ...     cron_expression="*/5 * * * *",
+                ...     description="A schedule just used for some temporary testing.",
+                ... )
 
+        """
+        _get_function_identifier(function_id, function_external_id)
+        nonce = _create_session_and_return_nonce(self._cognite_client, client_credentials)
```

Comment:
Most likely not a big issue, but previously, a user on GCP could create a schedule linked with `function_external_id` and without client credentials. The use case for this was primarily for functions with API-keys attached, but it could also be functions without API-keys attached (that just did some logic without needing to authenticate to CDF). Functions of the latter kind can no longer be created after this change.